engine/src/main/java/net/sparkzz/entropy/input/mappings/InputAction.java

```diff
@@ -0,0 +1,46 @@
+package net.sparkzz.entropy.input.mappings;
+
+import static org.lwjgl.glfw.GLFW.*;
+
+/**
+ * Enum representing input actions in the Entropy Engine.
+ * Each action is associated with a default GLFW key code.
+ *
+ * @author Brendon Butler
+ * @version 0.1.0-PREALPHA
+ * @since 2025-07-15
+ */
+public enum InputAction {
```

Comment:
This should likely be moved to the game logic since games utilizing this engine may have a different control scheme and custom/user bindings are for when users change the defaults in the settings.

In order to do this, the InputMapper should be genericized so that it can take in a custom Enum definition, something like this:
```java
// InputMapper.java
public class InputMapper<T extends Enum<T>> { /* ... */ }
```
```java
// Implementing game class
InputMapper<GameInputAction> gameMapper = new InputMapper<>(GameInputAction.class);
```

Reply:
@coderabbitai Do you agree with this suggestion and are there any other considerations I should make along these lines?

I would like to make the engine dynamic and easily extendable. For that reason, I believe moving InputAction enum to the game logic and making the InputMapper class generic based on the InputAction type would be a better approach. Alternatively, I can define a default InputAction enum in the engine, but still make the InputMapper generic so that anyone using the engine can either use the default InputAction enum for a lower barrier to entry, or create a custom enum for game-specific control schemes.